sklearn/neural_network/extreme_learning_machines.py

```diff
@@ -0,0 +1,629 @@
+            Subset of training data.
+
+        y : array-like, shape (n_samples,)
+            Subset of target values.
+
+        sample_weight : array-like, shape (n_samples,)
+            Per-sample weights. Rescale C per sample. Higher weights
+            force the classifier to put more emphasis on these points.
+
+        Returns
+        -------
+        self : returns a trained ELM usable for prediction.
+        """
+        self._fit(X, y, sample_weight=sample_weight, incremental=True)
+
+        return self
+
+    def _decision_scores(self, X):
+        """Predict using the ELM model
+
+        Parameters
+        ----------
+        X : {array-like, sparse matrix}, shape (n_samples, n_features)
+            The input data.
+
+        Returns
+        -------
+        y_pred : array-like, shape (n_samples,) or (n_samples, n_outputs)
+            The predicted values.
+        """
+        X = check_array(X, accept_sparse=['csr', 'csc', 'coo'])
+
+        if self.batch_size is None:
+            hidden_activations = self._compute_hidden_activations(X)
+            y_pred = safe_sparse_dot(hidden_activations, self.coef_output_)
+        else:
+            n_samples = X.shape[0]
+            batches = gen_batches(n_samples, self.batch_size)
+
+            y_pred = np.zeros((n_samples, self.n_outputs_))
+            for batch in batches:
+                h_batch = self._compute_hidden_activations(X[batch])
+                y_pred[batch] = safe_sparse_dot(h_batch, self.coef_output_)
+
+        return y_pred
+
+
+class ELMClassifier(BaseELM, ClassifierMixin):
+    """Extreme learning machine classifier.
+
+    The algorithm trains a single-hidden layer feedforward network by computing
+    the hidden layer values using randomized parameters, then solving
+    for the output weights using least-square solutions. For prediction,
+    after computing the forward pass, the continuous output values pass
+    through a gate function converting them to integers that represent classes.
+
+    This implementation works with data represented as dense and sparse numpy
+    arrays of floating point values for the features.
+
+    Parameters
+    ----------
+    C : float, optional, default 100
+        A regularization term that controls the linearity of the decision
+        function. Smaller value of C makes the decision boundary more linear.
+
+    class_weight : {dict, 'auto', None}, default None
+        If 'auto', class weights will be given inversely proportional
+        to the frequency of the class in the data.
+        If a dictionary is given, keys are the class labels and the
+        corresponding values are the class weights.
+        If None is given, then no class weights will be applied.
+
+    weight_scale : float, default 1.
+        Initializes and scales the input-to-hidden weights.
+        The weight values will range between plus and minus
+        'sqrt(weight_scale * 6. / (n_features + n_hidden))' based on the
```

Comment:
ah it is documented, nice :) still it would be interesting to know if that increases or decreases interactions between n_hidden and weight_scale.